### PR TITLE
feat(consent-manager)!: convert to CSS modules

### PR DESCRIPTION
**Pull request by zchsh**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1199185162145262/f)
🔍 [Preview Link](https://react-components-git-zsconsent-manager-css-modules-hashicorp.vercel.app/components/consentmanager)

---

This PR updates `@hashicorp/react-consent-manager` to use CSS modules.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: .changeset/rich-bugs-heal.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@hashicorp/react-consent-manager': major
+---
+
+- 💥✨ BREAKING CHANGE: Converts to CSS modules.
+  - Consumers will need to remove any `@hashicorp/react-consent-manager/style.css` imports.
+  - We've retained a `id="g-consent-manager` on the root element.
```
Comment (zchsh) [inline .changeset/rich-bugs-heal.md:7]: Added a debt task for this:
https://app.asana.com/0/1100423001970639/1200916857009754/f

File: packages/consent-manager/index.js [hunk 2]
```diff
@@ -80,7 +82,7 @@ export default class ConsentManager extends Component {
 
   render() {
     return (
-      <div id="g-consent-manager">
+      <div className={classNames(s.root, this.props.className)}>
```
Comment (zchsh) [inline packages/consent-manager/index.js:85]: Added support for `className` to ensure consumers can override hiding from Percy if needed.

File: pages/global.css [hunk 3]
```diff
@@ -12,7 +12,6 @@
 @import '../packages/enterprise-alert/style.css';
 @import '../packages/featured-slider/style.css';
 @import '../packages/content-cta/style.css';
-@import '../packages/consent-manager/style.css';
```
Comment (zchsh) [inline pages/global.css:15]: 🔪 🎉

File: packages/consent-manager/style.module.css [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.root {
+  @media only percy {
```
Comment (zchsh) [inline packages/consent-manager/style.module.css:2]: Closes out [this task](https://app.asana.com/0/1100423001970639/1200759443625270/f) and does the half the required work for [this task](https://app.asana.com/0/1100423001970639/1200916857009754/f).

**Review by BRKalow — APPROVED** [review]
LGTM!

**Pull request merged**